lib/benchmark/linear.go

```diff
@@ -0,0 +1,71 @@
+package benchmark
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/gravitational/teleport/lib/client"
+)
+
+// Linear generator
+type Linear struct {
+	LowerBound          int
+	UpperBound          int
+	Step                int
+	MinimumMeasurements int
+	MinimumWindow       time.Duration
+	currentRPS          int
+	config              Config
+}
+
+// Generate advances the Generator to the next generation.
+// It returns false when the generator no longer has configurations to run.
+func (l *Linear) Generate() bool {
+	if l.currentRPS < l.LowerBound {
+		l.currentRPS = l.LowerBound
+		return true
+	}
+	l.currentRPS += l.Step
+	return l.currentRPS <= l.UpperBound
+}
+
+// GetBenchmark returns the benchmark config for the current generation.
+// If called after Generate() returns false, this will result in an error.
+func (l *Linear) GetBenchmark() (context.Context, Config, error) {
+	if l.currentRPS > l.UpperBound {
+		return nil, Config{}, errors.New("No more generations")
+	}
+
+	return context.TODO(), Config{
+		MinimumWindow:       l.MinimumWindow,
+		MinimumMeasurements: l.MinimumMeasurements,
+		Rate:                l.currentRPS,
+		Threads:             10,
```

Comment:
Shouldn't `Threads` also come from `l`.

Also, `l` is a tough variable.  Looks kinda like I or 1.  `lg`?